internal/brokers/broker.go

```diff
@@ -161,6 +161,8 @@ func (b Broker) IsAuthenticated(ctx context.Context, sessionID, authenticationDa
 	case <-ctx.Done():
 		b.cancelIsAuthenticated(ctx, sessionID)
 		<-done
+		log.Warningf(ctx, "Authentication aborted: PAM client disconnected unexpectedly (session %s)", sessionID)
+		return auth.Cancelled, "{}", nil
```

Comment:
New error seems fine, but it's unclear to me why after having waited for `done` we don't return here what it was set in the call on line 148, where such values should be set by that... Or are they empty now because of the new error?

Reply:
well, if we want to handle the error in this function and not in the caller, we should not return the error. we might want to check if the error is indeed `context.Canceled`, which should be expected. I wonder how we should handle the case that the error is *not* `context.Canceled`, for example if broker already successfully authenticated before the `CancelIsAuthenticated` cancelled the authentication.